linera-indexer/src/graphql.rs

```diff
@@ -91,8 +93,14 @@ impl From<block::BlockBlockValueExecutedBlock> for ExecutedBlock {
     }
 }
 
-impl From<block::BlockBlock> for HashedValue {
-    fn from(val: block::BlockBlock) -> Self {
-        HashedValue::new_confirmed(val.value.executed_block.into())
+impl TryFrom<block::BlockBlock> for HashedValue {
+    type Error = IndexerError;
+    fn try_from(val: block::BlockBlock) -> Result<Self, Self::Error> {
+        match val.value.status.as_str() {
+            "validated" => Ok(HashedValue::new_validated(val.value.executed_block.into())),
+            // this constructor needs the "test" feature from linera-service
```

Comment:
Then let's remove that from `new_confirmed`. Let's not make production code depend on the `test` feature.

Reply:
I cannot remove it, the hashes are different between validated and confirmed blocks. Maybe the hash of the 'certificate' should only come from the executed block and not from the certificate value.

Reply:
Sorry, I meant we should remove the `#[cfg(features = "test")]` from `new_confirmed`.